### PR TITLE
CC-1140: Automated UI tests for account creation, roles, tools

### PR DESCRIPTION
**Pull request by cesarvh**

### CODE REVIEW

**Comment by cesarvh** [comment]
@pfarestveit Currently the credentials aren't hidden. Not too sure how to handle this, though since Admin is open to the world, I didn't think it would be a problem for core to have these additional users as well. Let me know if I should change this

File: pages/core/core_tools/core_tools_page.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def term_lists_link; {:xpath => '//a[contains(.,"Term Lists")]'} end
   def reports_link; {:xpath => '//a[contains(.,"Reports")]'} end
   def batch_link; {:xpath => '//a[contains(.,"Data Updates")]'} end
   def run_button; {:name => 'run'} end
-  def filter_bar; {:xpath => '//div[contains(@class, "AdminSearchBar")]//input[contains(@class,"LineInput")]'} end
+  # def filter_bar; {:xpath => '//div[contains(@class, "AdminSearchBar")]//input[contains(@class,"LineInput")]'} end
```
Comment (pfarestveit) [inline pages/core/core_tools/core_tools_page.rb:15]: Hey @cesarvh , did you intend to comment this out?

File: pages/core/core_tools/core_tools_page.rb [hunk 2]
```diff
@@ -41,10 +41,10 @@ def click_cancel_modal_button
     wait_for_element_and_click cancel_modal_button
   end
 
-  def fill_filter_bar(value)
-    wait_for_element_and_type(filter_bar, value)
-    sleep 1
-  end
+  # def fill_filter_bar(value)
```
Comment (pfarestveit) [inline pages/core/core_tools/core_tools_page.rb:44]: Same here

File: spec_helper.rb [hunk 3]
```diff
@@ -26,6 +26,7 @@
 require_relative 'models/user_role'
 require_relative 'models/user'
 require_relative 'models/data/core/core_tools/core_invocables_data.rb'
+# require_relative 'models/data/core/core_tools/core_invocables_data'
```
Comment (pfarestveit) [inline spec_helper.rb:29]: Also here

File: pages/pahma/pahma_procedures/pahma_use_of_collections_page.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class PAHMAUseOfCollectionsPage < CoreUCBUseOfCollectionsPage
 
-  include PAHMAUseOfCollectionsInfoForm
+  # include PAHMAUseOfCollectionsInfoForm
```
Comment (pfarestveit) [inline pages/pahma/pahma_procedures/pahma_use_of_collections_page.rb:3]: And here.

File: pages/pahma/pahma_objects/pahma_object_page.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@ class PAHMAObjectPage < CoreUCBObjectPage
   include CollectionSpacePages
   include PAHMAPages
   include PAHMASidebar
-  include PAHMAObjectIdInfoForm
+  # include PAHMAObjectIdInfoForm
```
Comment (pfarestveit) [inline pages/pahma/pahma_objects/pahma_object_page.rb:10]: And here

**Comment by pfarestveit** [comment]
Looks great, @cesarvh .  Regarding the credentials, I think it would be a good idea to put real credentials outside the code in case we want to be able to run the tests on our museum tenants at some point.  That setup already exists for museum admins:
* In config/settings.yml, add something like `test_user` under the `admin` lines for whatever deployments and whatever number of test users you want, with username and password as `secret`.
* In the override file for settings.yml, create the same settings lines with the real credentials.  Lines 5 thru 8 in config/config.rb show where the code looks for the override file.  
* In config/config.rb, add methods like the `Config.admin_user` method in order to initialize users.  If done like the admin method, it should import username and password from an override file if it exists.
* In config/test_config.rb, add the new users to the `get_users` method
* In the specs, you can invoke `get_users` on your `test_run` object to get all these user objects along with their credentials

If you want, you could also use models/user_role.rb to define distinct user roles, but that could always be done some other time.  Also, since this only runs on Lyrasis Core right now, I'm also fine merging without hiding the credentials as described above.  Just let me know what you want to do.

**Pull request merged**